pkg/reader/helpers.go

```diff
@@ -80,3 +80,13 @@ func validateReaderExistence[T contractreader.ContractReaderFacade](
 	}
 	return nil
 }
+
+func validateAccessorExistence(
+	accessors map[cciptypes.ChainSelector]cciptypes.ChainAccessor,
+	chainSelector cciptypes.ChainSelector,
+) error {
+	if _, exists := accessors[chainSelector]; !exists {
+		return fmt.Errorf("chain accessor not found for chain %d", chainSelector)
+	}
+	return nil
+}
```

Comment:
nit: you can avoid all the `r.accessors[sourceChainSelector]` calls by changing this to a getter instead of a validator.
```suggestion
func getAccessor(
	accessors map[cciptypes.ChainSelector]cciptypes.ChainAccessor,
	chainSelector cciptypes.ChainSelector,
) (cciptypes.ChainAccessor, error) {
	if accessor, exists := accessors[chainSelector]; exists {
		return accessor, nil
	}
	return nil, fmt.Errorf("chain accessor not found for chain %d", chainSelector)
}
```

Reply:
haha did this at first then decided to stick to the current pattern, I like this approach more though, created NONEVM-1995 to follow-up with this